### DISCUSSION TITLE
Simple Filter on a field

### DISCUSSION THREAD

**Discussion by BillyMcSkintos** [johnkerl/miller#1403] [Q&A] [answered]
i'm sure its something silly but I can't seem to apply a simple filter on column of a CSV:

I want to include only the rows where DEPARTMENTID = 9152::1

`C:\Apps\miller-6.9.0\mlr.exe --icsv --ocsv --from a.csv filter '$[DEPARTMENTID] =~ "^9152::1$"'  > b.csv`

outputs:
`mlr: cannot parse DSL expression.
Parse error on token "^" at line 1 column 20.
Expected one of:
  { ( field_name $[ braced_field_name $[[ $[[[ full_srec oosvar_name @[ braced_oosvar_name
  full_oosvar all non_sigil_name float int + - .+ .- ! ~ string_literal regex_case_insensitive
  int_literal float_literal boolean_literal null_literal inf_literal nan_literal
  const_M_PI const_M_E panic [ ctx_IPS ctx_IFS ctx_IRS ctx_OPS ctx_OFS ctx_ORS
  ctx_FLATSEP ctx_NF ctx_NR ctx_FNR ctx_FILENAME ctx_FILENUM env func`

**Comment by johnkerl** [comment, marked as answer]
Hi @BillyMcSkintos !

I had a very hard time getting all DSL expressions past the Windows terminal parser. There are some things that just can't be gotten through -- and this is a Windows issue; it has to do with how the command line has been mangled as soon as I get it from the OS. :(

Here are a couple workarounds -- please let me know if either of them helps:
https://miller.readthedocs.io/en/latest/miller-on-windows/#differences

**Reply by BillyMcSkintos** [reply]
Hi @johnkerl 

```
A third workaround is to replace " with """, then ' with ":
``` 

Am I replacing a single quote with a double quote or 2x single quotes? 
' = " or ''
Similarly is the double quote replaced with 3 double quotes or some combo of doubles and singles?
''''''''
"""

**Reply by BillyMcSkintos** [reply]
I went the file route instead and that worked.

`C:\Apps\miller-6.9.0\mlr.exe --icsv --ocsv --from a.csv filter -f filter1.mlr > b.csv`
